R/recycle.R

```diff
@@ -1,31 +1,126 @@
-#' Vector recycling
+#' Recycling rules used by r-lib and the tidyverse
+#'
+#' @description
+#' Recycling describes the concept of repeating elements of one vector to match
+#' the size of another vector. There are two rules that guide the recycling
+#' process used by packages in r-lib and the tidyverse:
+#'
+#' - Vectors of size 1 will be recycled to the size of any other vector
+#'
+#' - Otherwise, all vectors must have the same size
+#'
+#' @section Examples:
+#'
+#' Vectors of size 1 are recycled to the size of any other vector:
+#'
+#' ```
+#' vec_recycle_common(1:3, 1)
```

Comment:
Maybe use a familiar function like `tibble()` or `mutate()` to demonstrate the rules? The audience for this doc page might be large right?